cip-00XX.md

```diff
@@ -0,0 +1,73 @@
+
+### Effects of fee adjustment on AT deployment and runtime costs
+
+In the following table typical registering costs as well as runtime ones are listed (all values are in Burst and numbers in parenthesis are the cost with the current AT fees):
+
+![AT cases](./AT-cases.png)
+
+
+Taking the assumption of one BURST at one USD as a possible future scenario, it is clear that the current AT fees can make most applications infeasible. In this scenario, a low complexity AT (see table above) would cost around 7 USD per run (i.e. per transaction received). A more complex case can cost around 2’000 USD per run. 
+
+
+### Spam Attacks are still costy
+
+The basic assumption is that the fee mechanism currently available is effective in preventing spam attacks exploiting standard transactions/messages/etc. Thus, AT Spam attacks need to be at least as costly as they would be if the attacker uses regular transactions.
+
+1. STEP_FEE: was reduced to a FEE_QUANT/10 == 0.000735. This should be enough to avoid spam attacks by indefinitely running code. Since each simple addition, subtraction, etc cost as much as one tenth of the minimum fee of an entire transaction. Another possible attack is to program the following AT: it has a very large number of variables (costly to register), but runs very few steps (perhaps a single one) and is put to sleep until the next block. This is already prevented because every time the sleep function is called at least an API fee is already charged (10*STEP_FEE).
+
+1. MAX_STEPS: increasing this number by a factor of 50  will only reduce the load on the system. Limiting the number of steps per block can only force to store intermediate AT variable states in the blockchain, which only take space without additional benefits for miners or nodes. Further, this limit can be considered redundant with the STEP_FEE. Since each step already has a fee, an AT programmer already has the incentive to keep the steps minimal. Finally, the system already has another protection, which put an AT in pause if the AT balance reaches zero. This already avoids any kind of infinite recursion.
+
+1. COST_PER_PAGE: this cost is being reduced to a FEE_QUANT*10 == 0.0735 per page (256 bytes) when a new AT is registered (or created). Large AT programs can take several pages taking then more space in the blockchain and then affecting nodes. Even with the proposed modification each page (256 bytes) cost more than a message transaction can cost. Further, a message transaction can contain much more than 256 bytes (up to 4x times more). Under another perspective, registering a large (complex) AT mean potential for a large number of steps. This is, however, already taken care of by the step fees.
+
+
+### Compatibility
+
+The change will require a hard-fork. Any ATs created before the fork will be subject to the previous set of fees and will operate without any change. Only ATs created after the fork will run under the new fees.
+
```

Comment:
**Summary**
Unlike other blockchains, BURST does not have its own colored coin for running ATs. Registering and running ATs is supported by fees and the AT balance in BURST. With the current fees for ATs (version 1) any substantial increase in BURST price can make AT applications infeasible. In this CIP a new fee table is proposed (version 2). The proposed change would take place as a hard-fork and possibly attract new developers and applications for Burstcoin.

Reply:
section was missing